channeld/channeld.c

```diff
@@ -1145,14 +1143,19 @@ static void send_commit(struct peer *peer)
 		if (feerate > max)
 			feerate = max;
 
-		if (!channel_update_feerate(peer->channel, feerate))
-			status_failed(STATUS_FAIL_INTERNAL_ERROR,
-				      "Could not afford feerate %u"
-				      " (vs max %u)",
-				      feerate, max);
+		if (feerate != channel_feerate(peer->channel, REMOTE)) {
+			u8 *msg;
 
-		msg = towire_update_fee(NULL, &peer->channel_id, feerate);
-		sync_crypto_write(peer->pps, take(msg));
+			if (!channel_update_feerate(peer->channel, feerate))
```

Comment:
Is my reading right, that if this condition fails, we are unable to afford the feerate but we still send `update_fee`?

Reply:
If that fails, we abort because something is badly wrong.

Patch is hard to read, resulting code is easier:


```
		u32 feerate, max = approx_max_feerate(peer->channel);

		feerate = peer->desired_feerate;

		/* FIXME: We should avoid adding HTLCs until we can meet this
		 * feerate! */
		if (feerate > max)
			feerate = max;

		if (feerate != channel_feerate(peer->channel, REMOTE)) {
			u8 *msg;

			if (!channel_update_feerate(peer->channel, feerate))
				status_failed(STATUS_FAIL_INTERNAL_ERROR,
					      "Could not afford feerate %u"
					      " (vs max %u)",
					      feerate, max);

			msg = towire_update_fee(NULL, &peer->channel_id,
						feerate);
			sync_crypto_write(peer->pps, take(msg));
		}
```